src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+      :appearance="buttonAppearance"
+      class="k-file-upload-btn"
+      data-testid="k-file-upload-button"
+      size="small"
+      @click="updateFile"
+      @keyup.enter="updateFile"
+    >
+      {{ buttonText }}
+    </KButton>
+    <span
+      v-if="appearance === 'file'"
+      class="display-name cursor-pointer"
+      @click="updateFile"
+      @keyup.enter="updateFile"
+    >
+      {{ fileValue ? fileValue : placeholder }}
+    </span>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ref, PropType } from 'vue'
+import KLabel from '@/components/KLabel/KLabel.vue'
+import KInput from '@/components/KInput/KInput.vue'
+import KButton from '@/components/KButton/KButton.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import { v1 as uuidv1 } from 'uuid'
+
+export default defineComponent({
+  name: 'KFileUpload',
+
+  components: {
+    KLabel,
+    KInput,
+    KButton,
+    KIcon,
+  },
+
+  props: {
+    labelAttributes: {
+      type: Object,
+      default: () => ({}),
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Test mode - for testing only, strips out generated ids
+     */
+    testMode: {
+      type: Boolean,
+      default: false,
+    },
+    help: {
+      type: String,
+      default: undefined,
+    },
+    buttonAppearance: {
+      type: String,
+      default: 'primary',
+    },
+    buttonText: {
+      type: String,
+      default: 'Select file',
+    },
+    fileModel: {
+      type: String,
+      default: undefined,
+    },
+    removable: {
+      type: Boolean,
+      default: false,
+    },
+    placeholder: {
+      type: String,
+      default: 'No file selected',
+    },
+    /**
+     * Set whether its file upload or image upload
+     */
+    appearance: {
+      type: String,
+      default: 'file',
+      validator: (value: string): boolean => {
+        return ['file', 'image'].includes(value)
+      },
+    },
+    accept: {
+      type: Array as PropType<string[]>,
+      required: true,
+    },
+    maxFileSize: {
+      type: Number,
+      default: null,
+    },
+    /**
+     * Set icon size
+     */
+    iconSize: {
+      type: String,
+      default: '26',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Set icon color
+     */
+    iconColor: {
+      type: String,
+      default: '',
+    },
+  },
+
+  emits: ['file-added', 'file-removed', 'file-upload-error'],
+
+  setup(props, { emit }) {
+    const customInputId = computed((): string => props.testMode ? 'test-file-upload-id-1234' : uuidv1())
+    const maximumFileSize = computed((): Number => props.appearance === 'file' ? 5242880 : 1000000)
+
+    const hasUploadError = ref(false)
+
+    // This holds the FileList
+    const fileInput = ref([]) as any
+    // To clear the input value after reset
+    const fileInputKey = ref(0)
+    // File fakepath
+    const fileValue = ref('') as any
```

Comment:
Why does this need to be typed as `any`? 
```suggestion
    const fileValue = ref<any>('')
```

Reply:
It throws this `ts error` if I remove `as any `: `Property 'property-name' does not exist on type 'never'.`